### PR TITLE
Closes AgileVentures/MetPlus_tracker#484

### PR DESCRIPTION
**Pull request by jeanwei**
- #application_list is removed from 'jobs_controller.rb'
- partial: 'jobs/applied_job_list' & 'jobs/job_application' are combined for #application_list rendering
- 'jobs_controller' will be independent of 'job_application_viewer'
- #applications is removed from 'jobs_controller.rb'
- 'job/applications.html.haml' will be included in 'job/show.html.haml'
- every paths that point to #applications will be replace by #show

### CODE REVIEW

File: app/views/jobs/_applied_job_list.html.haml [hunk 1]
```diff
@@ -36,12 +43,44 @@
...
+                %td
+                  = link_to job_application.job_seeker.full_name,
+                            application_path(job_application.job_seeker)
+              - when :js_status
+                %td
+                  = job_application.job_seeker.job_seeker_status.short_description
+              - when :updated_at
+                %td
+                  - applied_at = job_application.status_change_time(:active)
+                  "#{time_ago_in_words(applied_at, )} ago"
```
Comment (patmbolger) [inline app/views/jobs/_applied_job_list.html.haml]: The field `:update_at` does not appear to be used (`:applied_at` is used in both field lists defined in JobApplicationsViewer#display_job_applications).

Please confirm and, if true, remove this field from this file.

File: app/views/jobs/_list_all.html.haml [hunk 2]
```diff
@@ -37,8 +37,6 @@
                         company_person_path(job.company_person))
           - when :num_applicants
             %td.col-md-1
-              = link_to_unless(job.number_applicants == 0,
-                               job.number_applicants,
-                               applications_job_path(job.id))
+              = job.number_applicants
```
Comment (jeanwei) [inline app/views/jobs/_list_all.html.haml]: currently this is done by clicking on job title
 
Previous view when click on job title:
image

Current view when click on job title:
image

Previous view when click on the number of applicants:
image

**Comment by jeanwei** [comment]
Thanks for the review. Code has updated as reviewed. I think by showing the views before and after would make this reviews simpler which I did not think of before this. Please let me know if you need more screenshots for certain views

File: app/views/jobs/_applied_job_list.html.haml [hunk 3]
```diff
@@ -36,12 +43,44 @@
               - when :applied_at
                 %td
                   - applied_at = job_application.status_change_time(:active)
-                  = "#{time_ago_in_words(applied_at)} ago"
+                  "#{time_ago_in_words(applied_at)} ago"
```
Comment (jeanwei) [inline app/views/jobs/_applied_job_list.html.haml]: changed

File: app/views/jobs/_applied_job_list.html.haml (hunk 1 above)
Comment (jeanwei) [inline app/views/jobs/_applied_job_list.html.haml]: removed

File: app/views/jobs/_list_all.html.haml (hunk 2 above)
Comment (patmbolger) [inline app/views/jobs/_list_all.html.haml]: OK, this makes sense - and is actually a better way to get access to the job applications for a job.

File: app/views/jobs/_applied_job_list.html.haml (hunk 3 above)
Comment (patmbolger) [inline app/views/jobs/_applied_job_list.html.haml]: I don't think the double quotes have been removed.  I still see them in the view.

screen shot 2016-11-13 at 8 45 28 am

**Comment by patmbolger** [comment]
I reviewed the PR.  I have one suggested change outstanding, and there are a few hound issues that should be addressed.

The screenshots help a lot.  I should have added those as part of my original comments.

**Comment by jeanwei** [comment]
I missed that out. double-quotes removed. 
exisiting view:
image

**Comment by jeanwei** [comment]
can we ignore line too long for this PR? i am working on remove long line too in the authorize job story

**Comment by patmbolger** [comment]
@jeanwei - yes, you can ignore "line too long".

**Pull request closed without merging**